src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs

```diff
@@ -184,6 +184,15 @@ private NamedTypeSymbol RetargetNamedTypeDefinition(NamedTypeSymbol type, Retarg
             {
                 Debug.Assert(type.IsDefinition);
 
+                if (type.IsNativeIntegerType)
+                {
+                    if (!SymbolMap.TryGetValue(type, out var retargeted))
```

Comment:
> if (!SymbolMap.TryGetValue(type, out var retargeted)) [](start = 19, length = 54)

I do not think the use of ```SymbolMap``` is warranted here. ```Retarget``` should unify the "underlying" type and ```AsNativeInteger``` is supposed to reuse instances itself.

Reply:
Also, it looks like using ```SymbolMap``` is not even appropriate here. It is for types defined in the compilation that needs retageting of type references. Native integers are not among the types, even their underlying types are not.

---
In reply to: [393248827](https://github.com/dotnet/roslyn/pull/42419#discussion_r393248827) [](ancestors = 393248827)